dataactcore/utils/jobQueue.py

```diff
@@ -36,52 +38,83 @@ def enqueue(jobID):
 
-                with open(timestamped_name, "w") as file:
-                    # get request
-                    response = requests.get(file_url)
-                    # write to file
-                    response.encoding = "utf-8"
-                    file.write(response.text)
+                full_file_path = "".join([CONFIG_BROKER['d_file_storage_path'], timestamped_name])
 
-                lines = []
-                with open(timestamped_name) as file:
-                    for line in reader(file):
-                        lines.append(line)
+                self.download_file(full_file_path, file_url)
+                lines = self.get_lines_from_csv(full_file_path)
 
                 headers = lines[0]
-                with CsvS3Writer(region, bucket, aws_file_name, headers) as writer:
+
+                if isLocal:
+                    file_name = "".join([CONFIG_BROKER['broker_files'], timestamped_name])
+                    csv_writer = CsvLocalWriter(file_name, headers)
+                else:
+                    file_name = "".join([str(user_id), "/", timestamped_name])
+                    bucket = CONFIG_BROKER['aws_bucket']
+                    region = CONFIG_BROKER['aws_region']
+                    csv_writer = CsvS3Writer(region, bucket, file_name, headers)
+
+                with csv_writer as writer:
                     for line in lines[1:]:
                         writer.write(line)
                     writer.finishBatch()
 
-                job_manager.setDFileStatus(d_file_id, "finished")
+                self.update_d_file_status(job_manager, d_file_id, "finished")
+                return {"message": "Success", "file_name": file_name}
             except Exception as e:
                 # Log the error
                 JsonResponse.error(e,500)
                 job_manager.setDFileMessage(d_file_id, str(e))
-                job_manager.setDFileStatus(d_file_id, "failed")
+                self.update_d_file_status(job_manager, d_file_id, "failed")
                 raise e
 
         self.enqueue = enqueue
         self.generate_d_file = generate_d_file
 
+    @staticmethod
+    def get_xml_response_content(api_url):
+        """ Retrieve XML Response from the provided API url """
+        return requests.get(api_url, verify=False).text
+
+    @staticmethod
+    def download_file(local_file_path, file_url):
+        """ Download a file locally from the specified URL """
+        with open(local_file_path, "w") as file:
+            # get request
+            response = requests.get(file_url)
+            # write to file
+            response.encoding = "utf-8"
+            file.write(response.text)
+
+    @staticmethod
+    def update_d_file_status(job_manager, d_file_id, status):
+        """ Update the D file status to the one specified via the Job Manager """
+        job_manager.setDFileStatus(d_file_id, status)
```

Comment:
I'd recommend staying away from static variables if you can.

Since we're mocking out `interface_holder` already (from which `job_manager` is derived), are we done?